lib/chef/encrypted_attribute/encrypted_mash/version0.rb

```diff
@@ -17,9 +17,20 @@
 # limitations under the License.
 #
 
+require 'chef/version'
 require 'chef/encrypted_attribute/encrypted_mash'
 require 'chef/encrypted_attribute/exceptions'
-require 'ffi_yajl'
+
+# Use the YAJL library that Chef provides.  Determine which to use based on
+# Chef version.
+major, minor = Chef::VERSION.split('.').take(2).map(&:to_i)
+if ([major, minor] <=> [11, 12]) == 1 # If [major, minor] > [11, 12]
+  require 'ffi_yajl'
+  YAJL_NAMESPACE = FFI_Yajl
+else
+  require 'yajl'
+  YAJL_NAMESPACE = Yajl
+end
```

Comment:
If you agree, I think it would be better to use the `Gem` class here:

```ruby
Gem::Requirement.new('< 11.13').satisfied_by?(Gem::Version.new(Chef::VERSION))
```

Your code is completely correct. But I think it is more difficult to read.

Also, it would be great if you could add some unit tests to compare the content of `YAJL_NAMESPACE`. Similar to those that you added in the cookbook.